src/lib/crypto/kimchi_pasta_snarky_backend/test/test_field.ml

```diff
@@ -30,6 +30,86 @@ module Make (Field : Kimchi_pasta_snarky_backend.Field.S_with_version) = struct
     let result_bits = Field.to_bits (Field.of_bits bs) in
     Alcotest.(check (list bool))
       "to_bits of_bits equivalence" expected_bits result_bits
+
+  let test_field_of_yojson_hex () =
+    let t = Field.random () in
+    let hex_json = Field.to_yojson t in
+    match Field.of_yojson hex_json with
+    | Ok t' ->
+        Alcotest.(check bool)
+          "of_yojson works with hex string" true (Field.equal t t')
+    | Error msg ->
+        Alcotest.fail (Printf.sprintf "Failed to parse hex JSON: %s" msg)
+
+  let test_field_of_yojson_decimal () =
+    let t = Field.random () in
+    let bigint = Field.to_bigint t in
+    let decimal_str = Field.Bigint.to_string bigint in
+    let decimal_json = `String decimal_str in
+    match Field.of_yojson decimal_json with
+    | Ok t' ->
+        Alcotest.(check bool)
+          "of_yojson works with decimal string" true (Field.equal t t')
+    | Error msg ->
+        Alcotest.fail (Printf.sprintf "Failed to parse decimal JSON: %s" msg)
+
+  let test_field_of_yojson_invalid_type () =
+    match Field.of_yojson (`Int 42) with
+    | Ok _ ->
+        Alcotest.fail "Should not parse non-string JSON"
+    | Error _ ->
+        Alcotest.(check bool)
+          "of_yojson should fail with non-string JSON" true true
+
+  let test_field_of_yojson_zero_decimal () =
+    (* Test parsing "0" as decimal string *)
+    match Field.of_yojson (`String "0") with
+    | Ok result ->
+        Alcotest.(check bool)
+          "Decimal 0 should equal Field.zero" true
+          (Field.equal result Field.zero)
+    | Error msg ->
+        Alcotest.fail (Printf.sprintf "Failed to parse 0 as decimal: %s" msg)
+
+  let test_field_of_yojson_one_decimal () =
+    (* Test parsing "1" as decimal string *)
+    match Field.of_yojson (`String "1") with
+    | Ok result ->
+        Alcotest.(check bool)
+          "Decimal 1 should equal Field.one" true
+          (Field.equal result Field.one)
+    | Error msg ->
+        Alcotest.fail (Printf.sprintf "Failed to parse 1 as decimal: %s" msg)
+
+  let test_field_of_yojson_42_decimal () =
+    (* Test parsing "42" as decimal string *)
+    match Field.of_yojson (`String "42") with
+    | Ok result ->
+        Alcotest.(check bool)
+          "Decimal 1 should equal Field.one" true
```

Comment:
copy paste typo in error message should say 42

Reply:
Fixed in bf4cb02af14cce2be8a9318418c40f803efb8378